src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -109,6 +109,14 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return manager;
 	}
 
+	public getNotebookManager(providerId: string): Promise<INotebookManager> {
```

Comment:
can we remove this down past the getters?

Reply:
Also, is there any reason why this needs to return a promise?

Reply:
This is for exception handle. I read some book suggesting for exception for this pattern. It is up to use to decide which patter we want to use.